crates/sui-types/src/object.rs

```diff
@@ -110,19 +122,28 @@ impl MoveObject {
     /// Contents of the object that are specific to its type--i.e., not its ID and version, which all objects have
     /// For example if the object was declared as `struct S has key { id: ID, f1: u64, f2: bool },
     /// this returns the slice containing `f1` and `f2`.
-    #[cfg(test)]
     pub fn type_specific_contents(&self) -> &[u8] {
         &self.contents[ID_END_INDEX..]
     }
 
     /// Update the contents of this object and increment its version
-    pub fn update_contents_and_increment_version(&mut self, new_contents: Vec<u8>) {
-        self.update_contents_without_version_change(new_contents);
+    pub fn update_contents_and_increment_version(
+        &mut self,
+        new_contents: Vec<u8>,
+    ) -> Result<(), SuiError> {
+        self.update_contents_without_version_change(new_contents)?;
         self.increment_version();
+        Ok(())
     }
 
     /// Update the contents of this object but does not increment its version
-    pub fn update_contents_without_version_change(&mut self, new_contents: Vec<u8>) {
+    pub fn update_contents_without_version_change(
```

Comment:
Seems like it's cleaner if this function just returns ExecutionError instead of SuiError

Reply:
Yep, good call. I thought that would create a cyclic dependency, but it should work.